theano/gpuarray/GpuMaxAndArgmax.py

```diff
@@ -0,0 +1,209 @@
+import os
+import numpy
+from six import integer_types
+import theano
+from theano.gof import Variable, Op, Apply
+from theano.tensor.type_other import NoneConst
+from theano.tensor.var import TensorConstant
+# from theano.tensor import as_tensor_variable
+from .basic_ops import (infer_context_name, as_gpuarray_variable)
+from .type import GpuArrayType
+
+try:
+    import pygpu
+except ImportError as e:
+    pass
+
+
+class GpuMaxAndArgmax(Op):
+    """
+    GPU version of MaxAndArgmax
+
+    """
+    __props__ = ()
+    argmax_dtype = "int64"
+
+    def make_node(self, X, axis=None):
+        context_name = infer_context_name(X)
+        # Check axis and convert it to Python variable.
+        if (isinstance(axis, (integer_types, numpy.integer)) or
+           (isinstance(axis, numpy.ndarray) and axis.ndim == 0)):
+            axis = [int(axis)]
+        elif isinstance(axis, (tuple, list, numpy.ndarray)):
+            axis = [int(a) for a in axis]
+        elif isinstance(axis, Variable):
+            if NoneConst.equals(axis):
+                axis = None
```

Comment:
Ok! I guess the wrapper method for max-and-argmax computation is ```def max_and_argmax(a, axis=None, keepdims=False)``` in theano/tensor/basic.py, isn't it ? So I'll transfer the axis parameter checking in that method.